src/client/Microsoft.Identity.Client/AuthenticationHeaderParser.cs

```diff
@@ -103,7 +103,7 @@ public static AuthenticationHeaderParser ParseAuthenticationHeaders(HttpResponse
 
                 if (wwwParameters.Any(parameter => parameter.AuthScheme == "PoP"))
                 {
-                    serverNonce = wwwParameters.Where(parameter => parameter.AuthScheme == "PoP").Single().Nonce;
+                    serverNonce = wwwParameters.Single(parameter => parameter.AuthScheme == "PoP").Nonce;
```

Comment:
avoid double iteration here. Both `Any` and `Single` will iterate the collection to apply the condition. 

You can do 

```
var serverNonce = wwwParameters.SingleOrDefault(parameter => parameter.AuthScheme == "PoP")?.Nonce
```